plugins/woocommerce/src/Blocks/Utils/ProductCollectionUtils.php

```diff
@@ -92,4 +99,171 @@ private static function remove_empty_array_recursive( $array ) {
 		}
 		return $array;
 	}
+
+	/**
+	 * Parse editor's location context from WP Post.
+	 *
+	 * Possible contexts:
+	 * - post
+	 * - page
+	 * - product
+	 * - product-archive
+	 * - cart
+	 * - checkout
+	 * - catalog
+	 * - order
+	 *
+	 * @param WP_Post $post The Post instance.
+	 *
+	 * @return string Returns the context.
+	 */
+	public static function parse_editor_location_context( $post ) {
+		$context = 'site';
+
+		if ( ! $post instanceof \WP_Post ) {
+			return $context;
+		}
+
+		$post_type = $post->post_type;
+		if ( ! in_array( $post_type, array( 'post', 'page', 'wp_template', 'wp_template_part' ), true ) ) {
+			return $context;
+		}
+
+		if ( in_array( $post_type, array( 'wp_template', 'wp_template_part' ), true ) ) {
+
+			$name = $post->post_name;
+			if ( false !== strpos( $name, SingleProductTemplate::SLUG ) ) {
+				$context = 'product';
+			} elseif ( ProductAttributeTemplate::SLUG === $name ) {
+				$context = 'product-archive';
+			} elseif ( false !== strpos( $name, 'taxonomy-' ) ) { // Including the '-' in the check to avoid false positives.
+				$taxonomy           = str_replace( 'taxonomy-', '', $name );
+				$product_taxonomies = get_object_taxonomies( 'product', 'names' );
+				if ( in_array( $taxonomy, $product_taxonomies, true ) ) {
+					$context = 'product-archive';
+				}
+			} elseif ( in_array( $name, array( CartTemplate::SLUG, MiniCartTemplate::SLUG ), true ) ) {
+				$context = 'cart';
+			} elseif ( CheckoutTemplate::SLUG === $name ) {
+				$context = 'checkout';
+			} elseif ( ProductCatalogTemplate::SLUG === $name ) {
+				$context = 'catalog';
+			} elseif ( OrderConfirmationTemplate::SLUG === $name ) {
+				$context = 'order';
```

Comment:
In the editor I used `order-confirmation`. Worth unifying.

```suggestion
				$context = 'order-confirmation';
```

I don't have strong opinion and if you feel `order` is better, please update in [here](https://github.com/woocommerce/woocommerce/blob/fd85199f9a27202ab25c6ca536a17189d2da8187/plugins/woocommerce-blocks/assets/js/blocks/product-collection/tracks-utils.ts#L11)

Reply:
Hmm, I think it's best to go with `order` here since there might be other areas related to the order later, such as the view order account page, an order's pay page, and possibly others. 

Good catch! I'll update the JS part as well, if you also agree!

Reply:
@kmanijak Now that I think about this more, I believe you might be right here. 

Since this context-matching logic will be used for tracking purposes only, having dedicated placements for the areas I mentioned above sounds like a good idea! 

Updated as per your suggestion [here](https://github.com/woocommerce/woocommerce/pull/46466/commits/775ac36adc725efaabd00d76629a5bb63e1ec536).